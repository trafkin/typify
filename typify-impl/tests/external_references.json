{
  "id": "https://schemas.mysite.com/schemas/external_references.json#",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "definitions": {
    "somename": {
      "properties": {
        "someproperty": {
          "$ref": "https://schemas.mysite.com/schemas/v1/external_reference.json#/definitions/someothername"
        }
      }
    }
  }
}