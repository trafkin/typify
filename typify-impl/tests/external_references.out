#[doc = r" Error types."]
pub mod error {
    #[doc = r" Error from a TryFrom or FromStr implementation."]
    pub struct ConversionError(std::borrow::Cow<'static, str>);
    impl std::error::Error for ConversionError {}
    impl std::fmt::Display for ConversionError {
        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
            std::fmt::Display::fmt(&self.0, f)
        }
    }
    impl std::fmt::Debug for ConversionError {
        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
            std::fmt::Debug::fmt(&self.0, f)
        }
    }
    impl From<&'static str> for ConversionError {
        fn from(value: &'static str) -> Self {
            Self(value.into())
        }
    }
    impl From<String> for ConversionError {
        fn from(value: String) -> Self {
            Self(value.into())
        }
    }
}
#[doc = "Somename"]
#[doc = r""]
#[doc = r" <details><summary>JSON schema</summary>"]
#[doc = r""]
#[doc = r" ```json"]
#[doc = "{"]
#[doc = "  \"properties\": {"]
#[doc = "    \"someproperty\": {"]
#[doc = "      \"$ref\": \"v1/external_reference/someothername\""]
#[doc = "    }"]
#[doc = "  }"]
#[doc = "}"]
#[doc = r" ```"]
#[doc = r" </details>"]
#[derive(Clone, Debug, serde :: Deserialize, serde :: Serialize)]
pub struct Somename {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub someproperty: Option<V1ExternalReferenceSomeothername>,
}
impl From<&Somename> for Somename {
    fn from(value: &Somename) -> Self {
        value.clone()
    }
}
#[doc = "V1ExternalReferenceSomeothername"]
#[doc = r""]
#[doc = r" <details><summary>JSON schema</summary>"]
#[doc = r""]
#[doc = r" ```json"]
#[doc = "{"]
#[doc = "  \"type\": \"object\","]
#[doc = "  \"properties\": {"]
#[doc = "    \"someproperty\": {"]
#[doc = "      \"a\": {"]
#[doc = "        \"type\": \"string\""]
#[doc = "      }"]
#[doc = "    }"]
#[doc = "  }"]
#[doc = "}"]
#[doc = r" ```"]
#[doc = r" </details>"]
#[derive(Clone, Debug, serde :: Deserialize, serde :: Serialize)]
pub struct V1ExternalReferenceSomeothername {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub someproperty: Option<serde_json::Value>,
}
impl From<&V1ExternalReferenceSomeothername> for V1ExternalReferenceSomeothername {
    fn from(value: &V1ExternalReferenceSomeothername) -> Self {
        value.clone()
    }
}
